include/roaring/roaring_types.h

```diff
@@ -97,6 +97,43 @@ typedef struct roaring_statistics_s {
     // and n_values_arrays, n_values_rle, n_values_bitmap
 } roaring_statistics_t;
 
+/**
+ *  (For advanced users.)
+ * The roaring64_statistics_t can be used to collect detailed statistics about
+ * the composition of a roaring64 bitmap.
+ */
+typedef struct roaring64_statistics_s {
+    uint32_t n_containers; /* number of containers */
+
+    uint32_t n_array_containers;  /* number of array containers */
+    uint32_t n_run_containers;    /* number of run containers */
+    uint32_t n_bitset_containers; /* number of bitmap containers */
+
+    uint32_t
+        n_values_array_containers;    /* number of values in array containers */
+    uint32_t n_values_run_containers; /* number of values in run containers */
+    uint32_t
+        n_values_bitset_containers; /* number of values in  bitmap containers */
+
+    uint32_t n_bytes_array_containers;  /* number of allocated bytes in array
+                                           containers */
+    uint32_t n_bytes_run_containers;    /* number of allocated bytes in run
+                                           containers */
+    uint32_t n_bytes_bitset_containers; /* number of allocated bytes in  bitmap
+                                           containers */
+
+    uint64_t
+        max_value; /* the maximal value, undefined if cardinality is zero */
+    uint64_t
+        min_value; /* the minimal value, undefined if cardinality is zero */
+    uint64_t sum_value; /* the sum of all values (could be used to compute
```

Comment:
I'm sad this is here: without it, we wouldn't have to visit every single value, only every container. And it's very possible the sum doesn't even fit in 64 bits. But it does match the statistics for the 32 bit bitmap.

I suppose, we could actually do something smarter for run containers than summing every value, but currently, we're not doing anything to compute the sum that the caller couldn't do equally as efficiently.

@lemire any thoughts on if it would be worth the inconsistency with 32 bit bitmaps to remove this?

Reply:
>  any thoughts on if it would be worth the inconsistency with 32 bit bitmaps to remove this?

I don't see why they have to match. 

I am not 100% sure why we have it for 32-bit bitmaps. Probably my fault.

Reply:
> I'm sad this is here: without it, we wouldn't have to visit every single value, 

We definitively should not visit every single value.

Reply:
sum was removed and and roaring64_bitmap_minimum, roaring64_bitmap_maximum was used to get the min and max